app.rb

```diff
@@ -0,0 +1,84 @@
+require_relative 'lib/database_connection'
+ require_relative 'lib/item_repository'
+ require_relative 'lib/order_repository'
+
+
+ class Application
+     def initialize(database_name, io, item_repository, order_repository)
+         DatabaseConnection.connect(database_name)
+         @io = io
+         @item_repository = item_repository
+         @order_repository = order_repository
+     end
+
+     def run
+         @io.puts "Welcome to the shop management program!"
+         @io.puts "What would you like to do?"
+         @io.puts "1 = List all shop items"
+         @io.puts "2 = Create a new item"
+         @io.puts "3 = List all orders"
+         @io.puts "4 = Create a new order"
+      
+         get_choice
+     end
+
+
+     private 
+
+     def get_choice 
+         input = @io.gets.chomp
+         fail "Please ensure input is either one of 1, 2, 3 or 4" unless input.to_i == 1 || input.to_i == 2 || input.to_i == 3 || input.to_i == 4 ||  input.to_i == 5 || input.to_i == 6
+         if input.to_i == 1
```

Comment:
I wonder if case / when might be a better or neater conditional here, but I don't know really.